core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -1212,17 +1212,15 @@ public void _testLikeAndSimilarFails() {
   }
 
   @Test void testCastRegisteredType() {
-    expr("cast(123 as customBigInt)")
-        .fails("class org.apache.calcite.sql.SqlIdentifier: CUSTOMBIGINT");
+    wholeExpr("cast(123 as customBigInt)")
+        .fails("Cast function does not support casting to type `CUSTOMBIGINT`");
     expr("cast(123 as sales.customBigInt)")
```

Comment:
Instead of message `Cast function does not support casting to type`, i guess `Cast from {fromType} to {toType} is not supported` is more clear ?

Reply:
Thanks for your review Danny. Based on the discussion in CALCITE-4265, I am not planning to move forward with this PR. Instead, I created a simpler PR at https://github.com/apache/calcite/pull/2326. Can you review that one instead? Thanks!